src/core/public/application/types.ts

```diff
@@ -33,22 +33,36 @@ import { RecursiveReadonly } from '../../utils';
 
 /** @public */
 export interface AppBase {
+  /**
+   * The unique identifier of the application
+   */
   id: string;
 
   /**
    * The title of the application.
    */
   title: string;
 
+  /**
+   * The status of the application.
+   */
+  status?: AppStatus;
+
+  /**
+   * An {@link AppStatusUpdater} observable that can be used to update
+   * the application {@link AppUpdatableFields} at runtime.
+   */
+  statusUpdater$?: Observable<AppStatusUpdater>;
```

Comment:
Since this can update tooltip or status, can we just call this `updater$`?

Reply:
Ok. I think I should rename `AppStatusUpdater` to `AppUpdater` and then `ApplicationService#registerAppStatusUpdater` to `registerAppUpdater` then?

Reply:
Sounds good to me.